eng/common/pipelines/templates/jobs/sdk-generation.yml

```diff
@@ -0,0 +1,200 @@
+      value: '22.13.x'
+    - name: PythonVersion
+      value: '3.13'
+    - name: SdkArtifactName
+      value: SDK_Artifact
+  pool:
+    name: $(LINUXPOOL)
+    vmImage: $(LINUXVMIMAGE)
+
+  steps:
+    - checkout: none
+
+    - pwsh: |
+        $urlPattern = '^https://github\.com/(?<organization>[^/]+)/(?<repository>[^/]+)'
+        if ('${{ parameters.SpecRepoUrl }}' -match $urlPattern) {
+          $specRepoOwner = $Matches['organization']
+          Write-Host "##vso[task.setvariable variable=SpecRepoOwner]$specRepoOwner"
+          Write-Host "SpecRepoOwner variable set to: $specRepoOwner"
+
+          $specRepoName = $Matches['repository']
+          Write-Host "##vso[task.setvariable variable=SpecRepoName]$specRepoName"
+          Write-Host "SpecRepoName variable set to: $specRepoName"
+
+          $specRepoDirectory = "$(System.DefaultWorkingDirectory)/$specRepoName"
+          Write-Host "##vso[task.setvariable variable=SpecRepoDirectory]$specRepoDirectory"
+          Write-Host "SpecRepoDirectory variable set to: $specRepoDirectory"
+        }
+
+        if ('${{ parameters.SdkRepoUrl }}' -match $urlPattern) {
+          $sdkRepoOwner = $Matches['organization']
+          Write-Host "##vso[task.setvariable variable=SdkRepoOwner]$sdkRepoOwner"
+          Write-Host "SdkRepoOwner variable set to: $sdkRepoOwner"
+
+          $sdkRepoName = $Matches['repository']
+          Write-Host "##vso[task.setvariable variable=SdkRepoName]$sdkRepoName"
+          Write-Host "SdkRepoName variable set to: $sdkRepoName"
+
+          $sdkRepoDirectory = "$(System.DefaultWorkingDirectory)/$sdkRepoName"
+          Write-Host "##vso[task.setvariable variable=SdkRepoDirectory]$sdkRepoDirectory"
+          Write-Host "SdkRepoDirectory variable set to: $sdkRepoDirectory"
+        }
+
+        if ([string]::IsNullOrEmpty($SpecRepoOwner) -or [string]::IsNullOrEmpty($SpecRepoName) -or [string]::IsNullOrEmpty($SdkRepoOwner) -or [string]::IsNullOrEmpty($SdkRepoName)) {
+          Write-Host "One or more required variables are null or empty. Ensure that SpecRepourl and SdkRepoUrl are set to valid GitHub repository URLs."
+          Exit 1
+        }
+
+        $sdkRepoNameForSpecGen = $sdkRepoName
+        if ($sdkRepoName.EndsWith('-for-net')) {
+          $sdkRepoNameForSpecGen = $sdkRepoName + "-track2"
+        }
+        Write-Host "##vso[task.setvariable variable=SdkRepoNameForSpecGen]$sdkRepoNameForSpecGen"
+        Write-Host "SdkRepoNameForSpecGen variable set to: $sdkRepoNameForSpecGen"
+      displayName: "Create Run Time Variables"
+
+    - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+      parameters:
+        Paths:
+        - '/*'
+        - '!sdk/**/test-recordings/*'
+        - '!sdk/**/recordings/*'
+        - '!sdk/**/SessionRecords/*'
+        - '!sdk/**/session-records/*'
+        Repositories:
+        - Name: $(SpecRepoOwner)/$(SpecRepoName)
+          Commitish: ${{ parameters.SpecRepoBranch }}
```

Comment:
@weshaggard seems current sparse-checkout script doesn't support checking out a specific commit from a specific branch. Should we extend it or add a step to the followed script step to checkout the specific commit?

Reply:
When you are checking out a commit you don't need to specify a branch it should just be the commit sha.

Reply:
All right, it would check out to the specific `commitish` when the value is not a branch. I think the script should be good for the release scenario as it would only depend on the first commit and doesn't need to worry about the previous commit.

Reply:
I believe the checkout of `commitsh` is the commit and all the history before it, @benbp right?

Reply:
Correct, it's the entire history from the commitish specified, whether it's a commit sha or a branch, tag, etc.

More info here: https://stackoverflow.com/a/23303550